Sources/Testing/Traits/Trait.swift

```diff
@@ -81,6 +81,40 @@ public protocol SuiteTrait: Trait {
   var isRecursive: Bool { get }
 }
 
+/// A protocol extending ``Trait`` that offers an additional customization point
```

Comment:
I want to do an editorial pass on this type's documentation after we promote it to API, but it's fine at this point.